### ISSUE TITLE
Enforce accepted values for the `type` attribute of the `<exclude-pattern>` and `<include-pattern>` tags

### ISSUE THREAD

**Issue by rodrigoprimo** [PHPCSStandards/PHP_CodeSniffer#727]
## Is your feature request related to a problem?

While reviewing https://github.com/PHPCSStandards/PHP_CodeSniffer/pull/705#pullrequestreview-2454541742, I noticed that in the code documentation, PHPCS specifies that there are two possible values for the `type` attribute of the `<exclude-pattern>` and `<include-pattern>` tags that can be used in a ruleset file: `absolute` and `relative`.

https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/9c8f30f9267c2f45952037c26392b5b32b7e971f/src/Ruleset.php#L48-L57
https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/9c8f30f9267c2f45952037c26392b5b32b7e971f/src/Ruleset.php#L59-L68

But the code does not validate that one of those values is passed and any value is accepted:

https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/9c8f30f9267c2f45952037c26392b5b32b7e971f/src/Ruleset.php#L1232
https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/9c8f30f9267c2f45952037c26392b5b32b7e971f/src/Ruleset.php#L1258

Adding the following to a ruleset.xml file, will add a exclude pattern (called ignore pattern in the code) with the type `InvalidValue`:

```xml
<exclude-pattern type="InvalidValue">./vendor/*</exclude-pattern>
```

## Describe the solution you'd like

PHPCS should validate the values passed to the `type` attribute and warn users if an invalid value is passed.

## Additional context (optional)

- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [ ] I intend to create a pull request to implement this feature.

**Comment by rodrigoprimo** [comment]
@jrfnl, I started investigating a solution for this issue, but found a few things that I would like to check with you before proceeding.

### Type for `Ruleset::$includePatterns`

I noticed that despite the code documentation for `Ruleset::$includePatterns` mentioning that the type of each entry can be either `absolute` or `relative`, it seems that the type is not utilized anywhere. Or at least, I was not able to find where it is used. 

As far as I can see, the type is used only for `Ruleset::$ignorePatterns` when processing global patterns in [src/Filters/Filter.php#L249](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/ad9a0b2df4c15f78e2b02808577d986968a7eb74/src/Filters/Filter.php#L249). The type is ignored when processing sniff-specific ignore and include patterns ([src/Ruleset.php#L1555-L1563](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/d8ace300eb1c8334de6687e5fec0351e1153d7bf/src/Ruleset.php#L1555-L1563)).

The wiki does mention that include patterns are always absolute (https://github.com/PHPCSStandards/PHP_CodeSniffer/wiki/Annotated-Ruleset):

> Please note that all sniff-specific include patterns are checked using absolute paths.

I wonder if, instead of enforcing the accepted values for the `type` attribute of the `include-pattern` tag, the code should be updated to remove, where possible, references to the type of the `include-pattern`? Maybe the ruleset XML schema should be updated to remove the `type` attribute from the `include-pattern` tag?

### Accepted values for the `type` attribute

When I opened this issue, I saw the code documentation mentioned in the issue description, which states that the value of the `type` attribute can be either `absolute` or `relative`. But now, checking the ruleset XML schema, I noticed that it [only accepts `relative` as the value for this attribute](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/8ff9283ab30cd296293b57039efe573aaf669862/phpcs.xsd#L88). `absolute` is used by default if no value is passed, but it is not accepted as an explicit value in the XML schema definition. I understand that it is not necessary to explicitly use `absolute` as it is the default value, but I fail to see the reason to not accept it. Should the XML schema definition be updated? If not, should this issue be updated to only accept `relative` as an explicit value for the `type` attribute?
